citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.jsp

```diff
@@ -27,13 +28,33 @@ $(function() {
 	$("#uriLoadingFoundEditor").popover();
 	$("#uriLoadingFailureEditor").popover();
 	
-	$("#submitForm").click(function() {
-		constructPersonArray("author");
-		constructPersonArray("editor");
+	$("#uriLoadingSpinnerCreator").hide();
+	$("#uriLoadingFailureCreator").hide();
+	$("#uriLoadingFoundCreator").hide();
+	
+	$("#uriLoadingFoundCreator").popover();
+	$("#uriLoadingFailureCreator").popover();
+	
+	$("#submitForm").click(function(e) {
+		constructPersonArray("author", "author", 0);
+		constructPersonArray("editor", "editor", 0);
+		var creatorSubmitCount = 0;
+		$(".creator").parent().closest("tr").each(function(idx, elem){
+			if($(elem).css("display")!="none") {
+				var ele = $(elem).children().first();
+				if(!(ele.attr("id")=="editor" || ele.attr("id")=="author")) {
+					constructPersonArray("creator", ele.attr("id"), creatorSubmitCount);
+					if($("." + ele.attr("id").toLowerCase() + "-item").length > 0) {
+						creatorSubmitCount = creatorSubmitCount + $("." + ele.attr("id").toLowerCase() + "-item").length;
```

Comment:
not sure what this is doing, let's talk about that tomorrow

Reply:
I'm doing this to have a running ID as we are calling constructPersonArray for each role. I'm not able to think of another way to iterate over different roles.